agents/supervisor/supervisor.go

```diff
@@ -268,6 +316,15 @@ const (
 	type_TEST_NOOP  agentpb.Type = 999 // built-in
 )
 
+func (s *Supervisor) changeLastState(agentID string, newState inventorypb.AgentStatus) {
+	s.arw.Lock()
+	defer s.arw.Unlock()
+	if s.agentStatuses == nil {
```

Comment:
This is not possible. After `s.agentStatuses` is set to `nil` in `stopAll()` this `Supervisor` instance should not be (and can't be reused). Let's remove this defensive measure because it only make it look like _maybe somehow_ this is possible.

Reply:
 I have added this because of that: https://travis-ci.com/percona/pmm-agent/jobs/191821047#L594
So there it's not `maybe`. Tests fail without it.

Reply:
That's a problem in the code that this `if` just hides.
`agentStatuses` should be set to `nil` only when all agents are fully stopped and all state changes are sent. There we assign it to `nil` only to allocate again.
Alternatively, we could not set `agentStatuses` to `nil` at all. It is done to other maps to prevent accidental `Supervisor` reuse. Since we can't reuse it already (other maps are assigned to `nil`), we can skip that assignment for `agentStatuses`.